sunpy/coordinates/wcs_utils.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+import astropy.wcs.utils
+
+from .frames import *
+
+__all__ = ['solar_wcs_frame_mapping']
+
+def solar_wcs_frame_mapping(wcs):
+    """
+    This function registers the coordinates frames to their FITS-WCS coordinate
+    type values in the `astropy.wcs.utils.wcs_to_celestial_frame` registry.
+    """
+
+    xcoord = wcs.wcs.ctype[0][0:4]
+    ycoord = wcs.wcs.ctype[1][0:4]
```

Comment:
it dosen't cover every case in this function, but it might be best to try that first, as it will handle the swapping.